scipy/optimize/_hessian_update_strategy.py

```diff
@@ -382,7 +385,7 @@ def _update_implementation(self, delta_x, delta_grad):
             z = delta_x
         # Do some common operations
         wz = np.dot(w, z)
-        Mw = self.dot(w)
+        Mw = self@w
```

Comment:
Note that `@` is an operator, so should have spaces around it (here and the 2 instances below).